discovery/gossiper.go

```diff
@@ -3443,3 +3443,41 @@ func (d *AuthenticatedGossiper) getPrematureMsgFromNode(pub [33]byte,
 
 	return messages, nil
 }
+
+// getPrematureMsgForAnn takes a channel announcement to fetch premature
+// messages for all related nodes.
+func (d *AuthenticatedGossiper) getPrematureMsgForAnn(peerPub [33]byte,
+	ann *lnwire.ChannelAnnouncement) ([]*processedNetworkMsg, error) {
+
+	// messages holds the premature messages to be reprocessed later.
+	messages := make([]*processedNetworkMsg, 0)
+
+	// nodeCacheKeySet is a set of the node's public keys. A set is used
+	// here as it's possible that the relaying node is the same node as
+	// the channel announcement's operating node.
+	nodeCacheKeySet := make(map[[33]byte]struct{})
+
+	// Add the nodes. Note that we also fetch the announcement's operating
+	// nodes in case there are premature messages saved too. For the
+	// messages saved in these caches, we could wait for these nodes to
+	// send the corresponding channel announcements and then reprocess
+	// them. However, we choose to process them here to converge the graph
+	// faster.
+	nodeCacheKeySet[peerPub] = struct{}{}
```

Comment:
I think you have to check the entire map of caches to be extra safe instead of only these 3 pubkeys